File: src/frontend/lib/model/person/person_dto.dart
import 'dart:convert';

import 'package:frontend/enumeration/document_type_enum.dart';
import 'package:frontend/enumeration/person_type_enum.dart';
import 'package:frontend/model/model.dart';

class PersonDTO implements Model {
  String? id;
  String? name;
  String? fantasyName;
  PersonTypeEnum? personType;
  String? document;
  DocumentTypeEnum? documentType;
  String? addressNumber;
  String? addressStreet;
  String? addressNeighborhood;
  String? addressCity;
  String? addressState;
  String? addressPostalCode;
  String? phone1;
  String? phone2;
  DateTime? createdAt;
  DateTime? updatedAt;
  String? createdBy;
  String? updatedBy;

  PersonDTO();

  @override
  void fillFromJson(Map<String, dynamic> map) {
    id = map['id'];
    name = map['name'];
    fantasyName = map['fantasyName'];
    personType = PersonTypeEnum.fromString(map["personType"] as String);
    document = map['document'];
    documentType = DocumentTypeEnum.fromString(map["documentType"] as String);
    addressNumber = map['addressNumber'];
    addressStreet = map['addressStreet'];
    addressNeighborhood = map['addressNeighborhood'];
    addressCity = map['addressCity'];
    addressState = map['addressState'];
    addressPostalCode = map['addressPostalCode'];
    phone1 = map['phone1'];
    phone2 = map['phone2'];
    createdAt = DateTime.parse(map['createdAt']);
    updatedAt = DateTime.parse(map['updatedAt']);
    createdBy = map['createdBy'];
    updatedBy = map['updatedBy'];
  }

  @override
  String toJson() {
    Map<String, dynamic> map = {};
    map.putIfAbsent("id", () => id);
    map.putIfAbsent("name", () => name);
    map.putIfAbsent("fantasyName", () => fantasyName);
    map.putIfAbsent("personType", () => personType!.name.toUpperCase());
    map.putIfAbsent("document", () => document);
    map.putIfAbsent("documentType", () => documentType!.name.toUpperCase());
    map.putIfAbsent("addressNumber", () => addressNumber);
    map.putIfAbsent("addressStreet", () => addressStreet);
    map.putIfAbsent("addressNeighborhood", () => addressNeighborhood);
    map.putIfAbsent("addressCity", () => addressCity);
    map.putIfAbsent("addressState", () => addressState);
    map.putIfAbsent("addressPostalCode", () => addressPostalCode);
    map.putIfAbsent("phone1", () => phone1);
    map.putIfAbsent("phone2", () => phone2);

    return jsonEncode(map);
  }

  @override
  String getId() {
    return id!;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/financeiro/dto/person/PersonDTO.java
package br.com.grupopipa.financeiro.dto.person;

import br.com.grupopipa.financeiro.dto.DTO;
import br.com.grupopipa.financeiro.entity.PersonEntity;
import br.com.grupopipa.financeiro.enumeration.DocumentTypeEnum;
import br.com.grupopipa.financeiro.enumeration.PersonTypeEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class PersonDTO implements DTO<PersonEntity> {

    private UUID id;
    private String name;
    private String fantasyName;
    private PersonTypeEnum personType;
    private String document;
    private DocumentTypeEnum documentType;
    private String addressNumber;
    private String addressStreet;
    private String addressNeighborhood;
    private String addressCity;
    private String addressState;
    private String addressPostalCode;
    private String phone1;
    private String phone2;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;

    @Override
    public UUID getId() {
        return this.id;
    }

    @Override
    public void fillFromEntity(PersonEntity entity) {
        this.id = entity.getId();
        this.name = entity.getName();
        this.fantasyName = entity.getFantasyName();
        this.personType = entity.getPersonType();
        this.document = entity.getDocument();
        this.documentType = entity.getDocumentType();
        this.addressNumber = entity.getAddressNumber();
        this.addressStreet = entity.getAddressStreet();
        this.addressNeighborhood = entity.getAddressNeighborhood();
        this.addressCity = entity.getAddressCity();
        this.addressState = entity.getAddressState();
        this.addressPostalCode = entity.getAddressPostalCode();
        this.phone1 = entity.getPhone1();
        this.phone2 = entity.getPhone2();
        this.createdAt = entity.getCreatedAt();
        this.updatedAt = entity.getUpdatedAt();
        this.createdBy = entity.getCreatedBy();
        this.updatedBy = entity.getUpdatedBy();
    }
}
